src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -33,6 +33,16 @@
   private static final String CONNECTION_URL_DOC =
       "List of Elasticsearch HTTP connection URLs e.g. ``http://eshost1:9200,"
       + "http://eshost2:9200``.";
+  public static final String CONNECTION_USERNAME_CONFIG = "connection.username";
+  private static final String CONNECTION_USERNAME_DOC =
+      "The username used to authenticate with Elasticsearch. "
+      + "The default is the empty string, and authentication will only be performed if "
+      + " both the username and password are non-empty.";
```

Comment:
We don't want to authenticate when the username is non-null, regardless of the password?

Reply:
You mean allow for an empty username or password?  It's a bad security practice.  Do you think some customers will want it?